src/interchange/src/avro/decode.rs

```diff
@@ -381,7 +381,7 @@ impl<'a, 'row> AvroDecode for AvroFlatDecoder<'a, 'row> {
         })?;
 
         let n = numeric::twos_complement_be_to_numeric(&mut buf, scale)
-            .map_err_to_string()
+            .map_err(|e| e.to_string_with_causes())
```

Comment:
```suggestion
            .map_err_to_string_with_causes()
```

Reply:
This doesn't work because `anyhow::Error` doesn't impl `std::error::Error`. I could change the trait bound on `ResultExt` to `AsRef<dyn Error>`, but that doesn't work for some other errors because there is no blanket `AsRef` impl and errors need to manually have that implemented (see https://doc.rust-lang.org/std/convert/trait.AsRef.html#reflexivity)

Reply:
Oof, I see, what a pain! Thanks for explaining.